online/src/app/store/reducers/cart.reducers.ts

```diff
@@ -0,0 +1,69 @@
+import { createReducer, on } from "@ngrx/store";
+import {
+    checkoutRequest, checkoutRequestError, checkoutRequestSucces, placeOrder, placeOrderSuccess, placeOrderError
+} from "../actions/cart.actions";
+import { initialCartState } from "../state/cart.state";
+
+export const cartReducer = createReducer(
+    initialCartState,
+
+    on(placeOrder, (state, { id }) => {
+        const placedOrder = state.productOrders.find(placedOrder => placedOrder.productId === id);
+        if (placedOrder === undefined) {
+            return {
+                ...state,
+                productOrders: [...state.productOrders, { productId: id, quantity: 1 }],
+                status: 'success',
+                error: "",
```

Comment:
Also, I would probably not put the error within the state. You only really need to show it once, in the error action. That's kinda it.